src/analysis/calculus/fderiv.lean

```diff
@@ -1914,6 +1914,59 @@ h.continuous.comp (continuous_const.prod_mk continuous_id)
 
 end bilinear_map
 
+namespace continuous_linear_equiv
+
+/-!
+### The set of continuous linear equivalences between two Banach spaces is open
+
+In this section we establish that the set of continuous linear equivalences between two Banach
+spaces is an open subset of the space of linear maps between them.  These facts are placed here
+because the proof uses `is_bounded_bilinear_map_comp.continuous`, proved just above as a consequence
+of its differentiability.
+-/
+
+protected lemma is_open [complete_space E] :
+  is_open {f : E →L[𝕜] F | ∃ e : E ≃L[𝕜] F, f = e} :=
+begin
+  by_cases h : nonempty (E ≃L[𝕜] F),
+  { have e := classical.choice h,
+    let O : (E →L[𝕜] F) → (E →L[𝕜] E) := λ f, (e.symm : F →L[𝕜] E).comp f,
+    cases subsingleton_or_nontrivial E with _i _i; resetI,
+    { convert is_open_univ,
+      ext f,
+      refine ⟨λ _, trivial, λ _, ⟨e, _⟩⟩,
+      ext x,
+      refine e.symm.injective _,
+      simp },
+    have h_O : continuous O,
+    { have h_e_symm : continuous (λ (x : E →L[𝕜] F), (e.symm : F →L[𝕜] E)) := continuous_const,
+      convert is_bounded_bilinear_map_comp.continuous.comp (continuous_id.prod_mk h_e_symm), },
+    convert units.is_open.preimage h_O using 1,
+    ext f',
+    split,
+    { rintros ⟨e', rfl⟩,
+      let w : units (E →L[𝕜] E) := continuous_linear_equiv.to_unit (e'.trans e.symm),
+      exact ⟨w, rfl⟩ },
+    { rintros ⟨w, hw⟩,
+      let e' : E ≃L[𝕜] E := continuous_linear_equiv.of_unit w,
+      use e'.trans e,
+      ext x,
+      have he'w : e' x = w x := rfl,
+      simp [hw, he'w] } },
+  { convert is_open_empty,
+    ext x,
+    refine ⟨λ h', _, λ h', empty_subset _ h'⟩,
+    obtain ⟨e, he⟩ := h',
+    exfalso,
+    exact not_nonempty_iff_imp_false.mp h e },
```

Comment:
This is the remaining proof I was unhappy with.  I was thinking of extracting the highlighted section as a lemma that in a not-nonempty topological space, every set is open.  (Maybe, that a not-nonempty topological space has discrete topology.). What do you think?  The other parts of the lemma might be golf-able.

Reply:
If you start with `is_open_iff_nhds`, then you have an element of your set immediately.

Reply:
I'm golfing this lemma.

Reply:
Done. I've reformulated this lemma using `range`. The new lemma is not defeq to the old one because of lhs/rhs in the equality. I hope, this won't cause too much trouble.

Reply:
It seems that the new `subsingleton` instances make a proof break. I'll fix it tonight.

Reply:
I've created #4511.

Reply:
Thanks for investigating this!